### PR TITLE
Implement `VhighBits` & `Vselect` for interpreter

### PR DESCRIPTION
**Pull request by dheaton-arm**
Implemented the following Opcodes for the Cranelift interpreter:
- `VhighBits` to reduce a vector to a scalar integer formed by
concatenating the MSB of each lane.
- `Vselect` to select lanes from two vectors controlled by a boolean
vector.

Copyright (c) 2021, Arm Limited

### CODE REVIEW

**Comment by afonso360** [comment]
This is what I see on CI.
```
 Caused by:
    0: Cranelift codegen error
    1: Verifier errors
    2: - inst2 (v0 = load.b8x16 v4): has an invalid controlling type b8x16
```

At a first glance i suspect it might be either #3306 or #3341.

I can investigate this a bit better later, but if it is #3306 it might be better to just rebase, if it is #3341 its a bit more complicated, since the solution on #3341 is wrong, and I am working on the correct solution using `bmask`, but its going to take a while...

**Comment by afonso360** [comment]
Its not a trampoline compile failure. `%vselect_p_i8x16` is failing the verifier after legalization. This is what comes out after the legalization pass (but before running the verifier).
```
function %vselect_p_i8x16(i64 ptr [%rcx], i64 ptr [%rdx], i64 ptr [%r8]) -> i8x16 [%xmm0] windows_fastcall {
                                block0(v4: i64, v5: i64, v6: i64):
[DynRexOp2fld#410]                  v0 = load.b8x16 v4
[DynRexOp2fld#410]                  v1 = load.i8x16 v5
[DynRexOp2fld#410]                  v2 = load.i8x16 v6
[DynRexMp3blend#910]                v3 = vselect v0, v1, v2
[Op1ret#c3]                         return v3
}
```

I'm not too familiar with the legalizer to be able to tell anything else.

**Comment by afonso360** [comment]
It looks like this is only failing on the legacy backend. The test files pass if we remove the `target x86_64 legacy haswell` line.

Since bytecodealliance/rfcs#12 is approved, I think it would be ok if we disabled this test file for the legacy backend. Or we could wait a few more days until the FCP window is closed. @cfallin suggestions?

**Comment by cfallin** [comment]
> Since bytecodealliance/rfcs#12 is approved, I think it would be ok if we disabled this test file for the legacy backend. Or we could wait a few more days until the FCP window is closed. @cfallin suggestions?

Yes, I agree, let's just exclude the old backend from this test.

File: cranelift/filetests/filetests/runtests/simd-vhighbits.clif [hunk 1]
```diff
@@ -0,0 +1,53 @@
+test interpret
+test run
+target aarch64
+set enable_simd
+target x86_64
```
Comment (afonso360) [inline cranelift/filetests/filetests/runtests/simd-vhighbits.clif:5]: This probably also needs a `machinst` so that it doesn't run on the legacy backend.

**Review by cfallin — APPROVED** [review]
Thanks!

**Pull request merged**